rule-engine/rule-engine-components/src/main/java/org/thingsboard/rule/engine/transform/TbCopyKeysNode.java

```diff
@@ -0,0 +1,113 @@
+        name = "copy keys",
+        configClazz = TbCopyKeysNodeConfiguration.class,
+        nodeDescription = "Copies the msg or metadata keys with specified key names selected in the list",
+        nodeDetails = "Will fetch fields values specified in list. If specified field is not part of msg or metadata fields it will be ignored." +
+                "If the msg is not a JSON object returns the incoming message as outbound message with <code>Failure</code> chain, " +
+                "otherwise returns transformed messages via <code>Success</code> chain",
+        uiResources = {"static/rulenode/rulenode-core-config.js"},
+        configDirective = "tbTransformationNodeCopyKeysConfig",
+        icon = "content_copy"
+)
+public class TbCopyKeysNode implements TbNode {
+
+    TbCopyKeysNodeConfiguration config;
+    List<Pattern> patternKeys = new ArrayList<>();
+    boolean fromMetadata;
+
+    @Override
+    public void init(TbContext ctx, TbNodeConfiguration configuration) throws TbNodeException {
+        this.config = TbNodeUtils.convert(configuration, TbCopyKeysNodeConfiguration.class);
+        this.fromMetadata = config.isFromMetadata();
+        config.getKeys().forEach(key -> {
+            this.patternKeys.add(Pattern.compile(key));
+        });
+    }
+
+    @Override
+    public void onMsg(TbContext ctx, TbMsg msg) throws ExecutionException, InterruptedException, TbNodeException {
+        TbMsgMetaData metaData = msg.getMetaData();
+        String msgData = msg.getData();
+        AtomicBoolean msgChanged = new AtomicBoolean(false);
+        JsonNode dataNode = JacksonUtil.toJsonNode(msgData);
+        if (dataNode.isObject()) {
+            if (fromMetadata) {
+                ObjectNode msgDataNode = (ObjectNode) dataNode;
+                Map<String, String> metaDataMap = metaData.getData();
+                metaDataMap.forEach((keyMetaData, valueMetaData) -> {
+                    if (checkKey(keyMetaData)) {
+                        msgChanged.set(true);
+                        msgDataNode.put(keyMetaData, valueMetaData);
+                    }
+                });
+                msgData = JacksonUtil.toString(msgDataNode);
+            } else {
+                dataNode.fields().forEachRemaining(entry -> {
+                    String keyData = entry.getKey();
+                    if (checkKey(keyData)) {
+                        msgChanged.set(true);
+                        metaData.putValue(keyData, JacksonUtil.toString(entry.getValue()));
+                    }
+                });
+            }
+        }
+        if (msgChanged.get()) {
+            ctx.tellSuccess(TbMsg.transformMsg(msg, msg.getType(), msg.getOriginator(), metaData, msgData));
+        } else {
+            ctx.tellSuccess(msg);
+        }
+    }
+
+    boolean checkKey(String key) {
+        Optional<Pattern> currentPattern = patternKeys.stream()
+                .filter(pattern -> pattern.matcher(key).matches())
```

Comment:
@YuriyLytvynchuk, probably the `anyMatch(Predicate<? super T> var1);` method will be better to use here. Review the Stream.class docs.